src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1325,36 +1325,10 @@ bool PolicyHandler::SaveSnapshot(const BinaryMessage& pt_string,
 
   return result;
 }
+#if !(defined(PROPRIETARY_MODE) || defined(EXTERNAL_PROPRIETARY_MODE))
 
-#ifdef EXTERNAL_PROPRIETARY_MODE
-void PolicyHandler::OnSnapshotCreated(
-    const BinaryMessage& pt_string,
-    const std::vector<int>& retry_delay_seconds,
-    int timeout_exchange) {
-  std::string policy_snapshot_full_path;
-  if (SaveSnapshot(pt_string, policy_snapshot_full_path)) {
-    MessageHelper::SendPolicyUpdate(policy_snapshot_full_path,
-                                    timeout_exchange,
-                                    retry_delay_seconds,
-                                    application_manager_);
-  }
-}
-#else  // EXTERNAL_PROPRIETARY_MODE
-void PolicyHandler::OnSnapshotCreated(const BinaryMessage& pt_string) {
-  LOG4CXX_AUTO_TRACE(logger_);
-  POLICY_LIB_CHECK_VOID();
-#ifdef PROPRIETARY_MODE
-  std::string policy_snapshot_full_path;
-  if (!SaveSnapshot(pt_string, policy_snapshot_full_path)) {
-    LOG4CXX_ERROR(logger_, "Snapshot processing skipped.");
-    return;
-  }
-  MessageHelper::SendPolicyUpdate(policy_snapshot_full_path,
-                                  policy_manager_->TimeoutExchange(),
-                                  policy_manager_->RetrySequenceDelaysSeconds(),
-                                  application_manager_);
-#else   // PROPRIETARY_MODE
-  LOG4CXX_ERROR(logger_, "HTTP policy");
+void PolicyHandler::OnSnapshotCreatedHTTPExt(const BinaryMessage& pt_string) {
```

Comment:
@LuxoftAKutsan 
Naming is non-informative, I think snapshot sending must be moved out and this method must be called like GetNextUrl